tests/foreman/api/test_product.py

```diff
@@ -104,12 +93,16 @@ def test_positive_update_1(self, attrs):
         @Feature: Product
 
         """
-        setattr(self.product, attrs[0], attrs[1])
-        self.product = self.product.update()
-        self.assertEqual(getattr(self.product, attrs[0]), attrs[1])
-
-    @data(*generate_strings_list())
-    def test_positive_update_2(self, new_name):
+        for attrs in valid_name_desc_list():
+            with self.subTest(attrs):
+                setattr(self.product, attrs.keys()[0], attrs.values()[0])
```

Comment:
Great to know this.  But I like to keep it as it is for more readability and to keep it with the same syntax as mentioned in https://docs.python.org/2/library/functions.html#setattr